### PR TITLE
Applied new azbi structure

### PR DESCRIPTION
**Pull request by atsikham**
* Applied new config structure with added vm_groups
* Added tests for updated configuration
* Increased versions

### CODE REVIEW

File: azbi/v0/azbi_v0.go [hunk 1]
```diff
@@ -13,22 +13,37 @@ import (
...
-	version = "v0.0.2"
+	version = "v0.1.0"
 )
 
 type Subnet struct {
 	Name            *string  `json:"name"`
 	AddressPrefixes []string `json:"address_prefixes"`
 }
 
+type Image struct {
```
Comment (mkyc) [inline azbi/v0/azbi_v0.go:24]: I'd propose to rename it to VmImage.

**Review by mkyc — CHANGES_REQUESTED** [review]
Just one comment.

File: azbi/v0/azbi_v0_test.go [hunk 2]
```diff
@@ -439,18 +654,638 @@ func TestConfig_Load(t *testing.T) {
...
+			}
+		}]
+	}
+}
+`),
+			want:    nil,
+			wantErr: &MinimalParamsValidationError{"one of vm groups is missing 'image.publisher' field or this field is empty"},
+		},
+		{
+			name: "empty vm_groups.image.publisher parameter",
```
Comment (mkyc) [inline azbi/v0/azbi_v0_test.go:1249]: This one is having the same name as in line 1042

File: azbi/v0/azbi_v0_test.go [hunk 3]
```diff
@@ -439,18 +654,638 @@ func TestConfig_Load(t *testing.T) {
...
+			}
+		}]
+	}
+}
+`),
+			want:    nil,
+			wantErr: &MinimalParamsValidationError{"one of vm groups is missing 'image.sku' field or this field is empty"},
+		},
+		{
+			name: "missing vm_groups.image.publisher parameter",
```
Comment (mkyc) [inline azbi/v0/azbi_v0_test.go:1215]: this one has the same name as in line 1008.

**Review by mkyc — CHANGES_REQUESTED** [review]

**Review by mkyc — APPROVED** [review]

**Review by erzetpe — APPROVED** [review]

File: azbi/v0/azbi_v0.go [hunk 4]
```diff
@@ -44,29 +59,42 @@ func NewConfig() *Config {
...
 }
 
-func (c *Config) Marshall() (b []byte, err error) {
+func (c *Config) Marshal() (b []byte, err error) {
 	//TODO validate that all required fields are filled
 	return json.MarshalIndent(c, "", "\t")
 }
 
-func (c *Config) Unmarshall(b []byte) (err error) {
+func (c *Config) Unmarshal(b []byte) (err error) {
```
Comment (seriva) [inline azbi/v0/azbi_v0.go:97]: Maybe something to consider later is to use some smarter way of validating the fields? Just first result on Google something like this:

https://github.com/go-validator/validator

Comment (mkyc) [reply azbi/v0/azbi_v0.go:97]: I'll add it as task for it. Good point.

**Pull request merged**